### PR TITLE
[SkyServe] Add CLI args for resources override in `sky serve {up, update}`

### PR DESCRIPTION
**Pull request by cblmemo**
A simplified version of #2879 . Only keep resources override.

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
```bash
$ sky serve up examples/serve/http_server/task.yaml --cpus 4+ --gpus A100
Service from YAML spec: examples/serve/http_server/task.yaml
Service Spec:
Readiness probe method:           GET /health
Readiness initial delay seconds:  20
Replica autoscaling policy:       Fixed 2 replicas        
Each replica will use the following resources (estimated):
I 02-04 17:27:38 optimizer.py:694] == Optimizer ==
I 02-04 17:27:38 optimizer.py:705] Target: minimizing cost
I 02-04 17:27:38 optimizer.py:717] Estimated cost: $3.7 / hour
I 02-04 17:27:38 optimizer.py:717] 
I 02-04 17:27:38 optimizer.py:840] Considered resources (1 node):
I 02-04 17:27:38 optimizer.py:910] ----------------------------------------------------------------------------------------------
I 02-04 17:27:38 optimizer.py:910]  CLOUD   INSTANCE        vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE     COST ($)   CHOSEN   
I 02-04 17:27:38 optimizer.py:910] ----------------------------------------------------------------------------------------------
I 02-04 17:27:38 optimizer.py:910]  GCP     a2-highgpu-1g   12      85        A100:1         us-central1-a   3.67          ✔     
I 02-04 17:27:38 optimizer.py:910] ----------------------------------------------------------------------------------------------
I 02-04 17:27:38 optimizer.py:910] 
Launching a new service 'sky-service-879a'. Proceed? [Y/n]: ^CAborted!
```

```bash
$ sky serve update dummy-service-name examples/serve/http_server/task.yaml --cpus 4+ --gpu
s A100
Service from YAML spec: examples/serve/http_server/task.yaml
Service Spec:
Readiness probe method:           GET /health
Readiness initial delay seconds:  20
Replica autoscaling policy:       Fixed 2 replicas        
New replica will use the following resources (estimated):
I 02-04 17:29:07 optimizer.py:694] == Optimizer ==
I 02-04 17:29:07 optimizer.py:705] Target: minimizing cost
I 02-04 17:29:07 optimizer.py:717] Estimated cost: $3.7 / hour
I 02-04 17:29:07 optimizer.py:717] 
I 02-04 17:29:07 optimizer.py:840] Considered resources (1 node):
I 02-04 17:29:07 optimizer.py:910] ----------------------------------------------------------------------------------------------
I 02-04 17:29:07 optimizer.py:910]  CLOUD   INSTANCE        vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE     COST ($)   CHOSEN   
I 02-04 17:29:07 optimizer.py:910] ----------------------------------------------------------------------------------------------
I 02-04 17:29:07 optimizer.py:910]  GCP     a2-highgpu-1g   12      85        A100:1         us-central1-a   3.67          ✔     
I 02-04 17:29:07 optimizer.py:910] ----------------------------------------------------------------------------------------------
I 02-04 17:29:07 optimizer.py:910] 
Updating service 'dummy-service-name'. Proceed? [Y/n]: ^CAborted!
```
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

**Comment by cblmemo** [comment]
@Michaelvll This is ready for another look 🫡

**Review by Michaelvll — COMMENTED** [review]
Thanks for the PR @cblmemo! We should get this PR in soon. Just one comment.

File: sky/cli.py [hunk 1]
```diff
@@ -4448,10 +4505,40 @@ def serve_up(
...
+        else:
+            service_name = name
+    else:
+        if name is None:
+            name = service_name
+        else:
+            raise click.UsageError(
+                'Cannot specify both --name and --service-name. '
+                'Both of them are used to specify the name of the service. '
+                f'Got: {name!r} and {service_name!r}.')
```
[the remark this answers is not in the document]
Comment (cblmemo) [reply sky/cli.py]: Good point! Done. Thanks!

File: sky/cli.py [hunk 2]
```diff
@@ -4349,15 +4351,54 @@ def serve():
...
     if not is_yaml:
         raise click.UsageError('SERVICE_YAML must be a valid YAML file.')
+    env = _merge_env_vars(env_file, env)
     # We keep nargs=-1 in service_yaml argument to reuse this function.
     task = _make_task_or_dag_from_entrypoint_with_overrides(
-        service_yaml_args, entrypoint_name='Service')
+        service_yaml_args,
+        # For Service YAML, we disabled the `name` field override as
+        # it might be confusing with `service_name` field.
+        name=None,
```
Comment (Michaelvll) [inline sky/cli.py:4384]: I guess it should be fine to override the job name with `service_name`?

**Pull request merged**